mllib/src/test/scala/org/apache/spark/ml/source/image/ImageFileFormatSuite.scala

```diff
@@ -104,6 +105,39 @@ class ImageFileFormatSuite extends SparkFunSuite with MLlibTestSparkContext {
     assert(firstBytes20Set === expectedFirstBytes20Set)
   }
 
+  test("recursive loading") {
+    val imagesPath = spark.read.format("image")
+      .option("dropInvalid", true)
+      .option("recursiveFileLookup", true)
+      .load(recursiveImagePath)
+      .select(col("image.origin"))
+      .collect()
+      .map { row =>
+        val path = row.getString(0)
+        val keyStr = "data/mllib/images/"
+        path.substring(path.indexOf(keyStr) + keyStr.length)
+      }
+
+    assert(Set(imagesPath: _*) === Set(
```

Comment:
Nit: this is not maintainable. If we add files under the folder  "data/mllib/images/", we need to update this test case. It would be great if we use some IO lib to get all the expected files under the folder instead of hardcoding.

Reply:
@gengliangwang What about remove the testcase in `ImageFileFormatSuite` ? The test in `FileBasedDatasourceSuite` is enough.